frontend/src/components/content-grid/preview/generic.vue

```diff
@@ -39,5 +51,15 @@ c-anchor-root.group.w-full(class="max-w-[416px]")
         | {{ viewership }} {{ viewershipName + (viewership === 1 ? '' : 's') }}
         | •
         | {{ dateToElapsedTimeString(beginAt) }}
-    c-icon.text-slate-300(v-if="icon" :name="icon")
+    .flex.flex-col.justify-between.items-center
+      template(v-if="kebabMenu")
+        c-dropdown.flex.justify-center(class="[&_>_ul]:w-fit [&_>_ul]:right-0 [&_>_ul]:top-full")
+          c-dropdown-item(class="[&_>_li]:!bg-frost-blue hover:[&_>_li]:!bg-frost-deep")
+            router-link.block(:to="{ name: 'edit', params: { videoId: videoId } }")
+              span Edit video
+              c-anchor-cover
+
+          template(#button="{ toggled }")
+            c-button.text-xl(class="!p-0 !px-2" :theme="toggled ? 'blue' : 'invisible'") ⋮
+      c-icon.text-slate-300(v-if="icon" :name="icon")
```

Comment:
I would convert this entire section into a template (name it 'adorners') to avoid leaking so much data to this generic component (which is also used for live previews), then fill it in in the user-video-preview component only when this is the user's personnal video page.